codecs/h264_packet_test.go

```diff
@@ -291,3 +292,35 @@ func TestH264Payloader_Payload_SPS_and_PPS_handling(t *testing.T) {
 		t.Fatal("SPS and PPS aren't packed together")
 	}
 }
+
+func TestH264Payloader_PayloadAndDetectDataIframeAfterSpsPps(t *testing.T) {
+	pck := H264Payloader{}
+
+	// When packetizing SPS and PPS are emitted with following NALU
+	res := pck.Payload(1500, []byte{0x07, 0x00, 0x01})
+	if len(res) != 0 {
+		t.Fatal("Generated payload should be empty")
+	}
+
+	res = pck.Payload(1500, []byte{0x08, 0x02, 0x03})
+	if len(res) != 0 {
+		t.Fatal("Generated payload should be empty")
+	}
+
+	idx := -1
+	res, idx = pck.PayloadAndDetectData(1500, []byte{0x05, 0x04, 0x05})
+
+	if idx != 1 {
+		t.Fatal("data index when nalu 5 starts after sps and pps must be 1")
+	}
+}
+
+func TestH264Payloader_PayloadAndDetectDataIframe(t *testing.T) {
+	pck := H264Payloader{}
+
+	_, idx := pck.PayloadAndDetectData(1500, []byte{0x05, 0x04, 0x05})
```

Comment:
How about giving it a bit more complex payload where there are additional NAL units (Which we know exist), especially where the iframe is not in the beginning instead of the simplest case where almost nothing can possibly go wrong ? (Even **multiple data units** to ensure you actualy get the index of the **first** one as intended)
e.g. single payload stream with SPS,PPS,IFRAME,IFRAME (Also expecting to have NAL start code (0x0, 0x0, 0x0, 0x1) before first NAL, although it might not be required... but it is expected I think in H264 byte stream)
So
{0x0, 0x0, 0x0, 0x1, 0x07, 0x00, 0x01, // SPS
 0x0, 0x0, 0x0, 0x1, 0x08, 0x02, 0x03 // PPS
 0x0, 0x0, 0x0, 0x1, 0x05, 0x04, 0x05, // IFRAME-1
 0x0, 0x0, 0x0, 0x1, 0x05, 0x04, 0x05, // IFRAME-2 (Doesn't matter if this may not be a valid scenario)
}
And verify data idx to be 2 of IFRAME-1

Can be done as a separate test if you want... But I think it can replace this simple test.